backend/internal/handlers/restaurant/service.go

```diff
@@ -157,3 +166,100 @@ func (s *Service) DeleteRestaurant(id primitive.ObjectID) error {
 	}
 	return nil
 }
+
+// GetRestaurantFriendsFav
+// average of all your friends reviews for all menu items at restaurant --> if above 4 its a friend fav
+// if its a friend fav -- display all of the friends that have been to the restaurant
+func (s *Service) GetRestaurantFriendsFav(uid primitive.ObjectID, rid primitive.ObjectID) (*FriendsFav, error) {
+	ctx := context.Background()
+
+	var restaurant RestaurantDocument
+	var friends []string
+	var totalRating int
+	var numOfRatings int
+	var avgRating float64
+	var friendsFav *FriendsFav
+	// restaurant is stored in doc
+	err := s.restaurants.FindOne(ctx, bson.M{"_id": rid}).Decode(&restaurant)
+	if err == mongo.ErrNoDocuments {
+		// No matching restaurant found
+		return nil, mongo.ErrNoDocuments
+	} else if err != nil {
+		// Different error occurred
+		return nil, err
+	}
+	for _, menuItemId := range restaurant.MenuItems {
+		// why are the inputs for this method strings when all id's are represented as ObjectID's ???
+		friendReviews, err := s.userConnectionsService.GetFriendReviewsForItem(uid.Hex(), menuItemId.Hex())
+		if err != nil {
+			return nil, err
+		}
+		for _, friendReview := range friendReviews {
+			// storing the friends of the user that reviewed menuItems at the restaurant
+			friend := friendReview.Reviewer.ID
+			// ensure that all the friends added to the list are unique
+			if doesNotContain(friends, friend) {
+				friends = append(friends, friend)
+			}
+			totalRating += friendReview.Rating.Overall
+			numOfRatings += 1
+		}
+	}
+	avgRating = float64(totalRating) / float64(numOfRatings)
+	// the restaurant is a friends fav
+	if avgRating >= 4 {
+		friendsFav = &FriendsFav{
+			IsFriendsFav:    true,
+			FriendsReviewed: len(friends),
+		}
+	} else {
+		friendsFav = &FriendsFav{
+			IsFriendsFav:    false,
+			FriendsReviewed: len(friends),
+		}
+	}
+
+	return friendsFav, nil
+}
+
+// checks if an element is not in a slice
+func doesNotContain(slice []string, item string) bool {
+	for _, element := range slice {
+		if element == item {
+			return false
+		}
+	}
+	return true
+}
+
+func (s *Service) GetSuperStars(rid primitive.ObjectID) (int, error) {
+	ctx := context.Background()
+
+	var restaurant RestaurantDocument
+	var superStars int
+
+	// restaurant is stored in doc
+	err := s.restaurants.FindOne(ctx, bson.M{"_id": rid}).Decode(&restaurant)
+	if err == mongo.ErrNoDocuments {
```

Comment:
Both cases are equal to "return 0, err" -> if theres no special logging for the specific mongo error then we could just do "if err != nil return err"